app/src/main/java/io/homeassistant/companion/android/qs/TileExtensions.kt

```diff
@@ -49,6 +50,8 @@ abstract class TileExtensions : TileService() {
 
     private val mainScope = MainScope()
 
+    private var stateUpdateJob: CoroutineScope? = null

```

Comment:
This is an entire scope, not a `Job`. Is there a reason not the use the `mainScope` for this, so we are sure that the job inside it is also cancelled when `mainScope` is cancelled for other reasons (like in `onDestroy`)?

This could be `private var stateUpdateJob: Job? = null` and it can be assigned by using `stateUpdateJob = mainScope.launch { ... }`.

Reply:
No you are right it should be a `Job` and it works as expected with your suggestion. Thank you 🙏